client/modules/IDE/actions/project.js

```diff
@@ -265,12 +265,12 @@ export function cloneProject(id) {
       if (!id) {
         resolve(getState());
       } else {
-        apiClient.get(`/projects/${id}`)
-          .then(res => res.json())
-          .then(data => resolve({
-            files: data.files,
+        const currentUser = getState().user.username;
```

Comment:
I don't think this is the right solution—what if a user is duplicating a sketch that doesn't belong to them?

Reply:
I see...but could you please elaborate on the `duplicating a sketch that doesn't belong to the user` part?

I went through the `GET` routes specified on the server side and the appropriate one appeared to be `router.get('/:username/projects/:project_id', ...)`, so I thought of implementing it this way.

Reply:
Sketches/projects (you can use the terms interchangeably) are unique by id so you shouldn't need to include the username. I don't think this is the reason it's failing—I would look at how it's different when calling `cloneProject` from `SketchList.jsx` versus `Nav.jsx`.

Reply:
I agree the route should remain `'/projects/:project_id'`, however, I couldn't find an implementation that would allow fetching of projects only by their ID. So I took the liberty of adding another controller and `GET` route for the same. 

Could you please take a look?